<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Dongyue Xie" />

<meta name="date" content="2018-10-04" />

<title>bsseq</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">smashgen</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">bsseq</h1>
<h4 class="author"><em>Dongyue Xie</em></h4>
<h4 class="date"><em>2018-10-04</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-10-05</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20180501)</code> </summary></p>
<p>The command <code>set.seed(20180501)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/DongyueXie/smash-gen/tree/877806a60937bac8a35ca1f655cceb87fdd20bb0" target="_blank">877806a</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  analysis/gam.Rmd
    Untracked:  analysis/literature.Rmd
    Untracked:  analysis/meeting1005.Rmd
    Untracked:  data/chipexo_examples/
    Untracked:  data/chipseq_examples/

Unstaged changes:
    Modified:   analysis/sigma.Rmd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/DongyueXie/smash-gen/blob/877806a60937bac8a35ca1f655cceb87fdd20bb0/analysis/bsseq.Rmd" target="_blank">877806a</a>
</td>
<td style="text-align:left;">
Dongyue Xie
</td>
<td style="text-align:left;">
2018-10-05
</td>
<td style="text-align:left;">
add files
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>I found a DNA methylation dataset from Hansen et al 2011. This dataset can be assessed through R pacakge <a href="https://www.bioconductor.org/packages/release/data/experiment/html/bsseqData.html">bsseqData</a>. A detailed tutorial on the dataset and method from Hansen et al 2011 is also avaiable <a href="http://bioconductor.org/packages/release/bioc/html/bsseq.html">here</a>.</p>
<p>There are total 6 samples where three are from normal and three are from cancer. Each sample has 958541 CpGs from Chromesome 21 and 22. There are roughly 136019 CpGs with no data at all in any of the 6 samples. There are 572628 CpGs which are covered by at least 1 read in all 6 samples. Smoothing is done separately for each sample, only using the data where the coverage (for that sample) is non-zero. This estimates a genome-wide methylation profile, which is then evaluated in all CpGs in the BSseq object. As a result, after smoothing, every CpG in the object has an estimated methylation value.</p>
<p>They use t-test to find differentially methylated regions(DMRs). Before computing t-statistics, CpGs with little or no coverage are removed.</p>
</div>
<div id="initial-try" class="section level2">
<h2>Initial try</h2>
<p>I tried to apply smashgen to this DNA methylation dataset.</p>
<ol style="list-style-type: decimal">
<li>The CpGs loci with no data at all in any of the 6 samples were removed. Should I?</li>
<li>Choose bin size s.t. the resulted binomial sequence length is a power of 2. The last interval constains less CpGs. Any impreovement on this?</li>
<li>In each bin, <span class="math inline">\(n_t\)</span> is the total coverage, <span class="math inline">\(x_t\)</span> is the total possible methylation. For example, coverages are 0,1,2,3,4; methylations are 0,0,1,3,1, then <span class="math inline">\(n_t=10\)</span>, <span class="math inline">\(x_t=5\)</span>. Should I remove no coverage area? If removed, how to compare diffwerent samples?</li>
<li>After obtaining binomial sequence, I do a reflection on both side then smooth the data.</li>
</ol>
<p>I may misunderstand it…Inside each window(at least 2000kb), they treat each base methylation follows binomial. So is Bionimial(<span class="math inline">\(n_{reads},p_{meth}\)</span>)???</p>
<pre class="r"><code>library(bsseqData)
library(bsseq)
data(&quot;BS.cancer.ex&quot;)

reflect=function(x,direct=&#39;left&#39;,len){
  n=length(x)
  if(direct==&#39;left&#39;){
    x=c(rev(x[1:len]),x)
  }
  if(direct==&#39;right&#39;){
    x=c(x,rev(x[(n-len+1):n]))
  }
  if(direct==&#39;both&#39;){
    x=c(rev(x[1:len[1]]),x,rev(x[(n-len[2]+1):n]))
  }
  return(x)
}


# remove no coverage bases for all 6 sampels
remove.idx=which(rowSums(getCoverage(BS.cancer.ex)) == 0)
result=list()
for (s in 1:6) {
  # Cancer 1 samples
BS1=BS.cancer.ex[,s]
BS1=BS1[-remove.idx]
# Obtain coverage
BS1.cov=getCoverage(BS1,type=&#39;Cov&#39;)
# Obtain possible methylation
BS1.M=getCoverage(BS1,type=&#39;M&#39;)
# choose 2kb windows. 
w=1607
nw=floor(length(BS1)/w)
# Obtain binomial data n_t and x_t
n_t=c()
x_t=c()
for (i in 1:nw) {
  n_t[i]=sum(BS1.cov[(w*(i-1)+1):(w*i)])
  x_t[i]=sum(BS1.M[(w*(i-1)+1):(w*i)])
}

n_T=sum(BS1.cov[-w*nw])
x_T=sum(BS1.M[-w*nw])
n_t=c(n_t,n_T)
x_t=c(x_t,x_T)

#bio.sm=smash_gen_lite(x_t,ntri=n_t,dist_family = &#39;binomial&#39;)
result[[s]]=list(n_t=n_t,x_t=x_t)
}

sm.ref=c()
for (k in 1:6) {
  nn=reflect(result[[k]]$n_t,direct = &#39;both&#39;,len = c(256,256))
  xx=reflect(result[[k]]$x_t,direct = &#39;both&#39;,len = c(256,256))
  ss=smash_gen_lite(xx,ntri=nn,dist_family = &#39;binomial&#39;)
  sm.ref=rbind(sm.ref,ss[257:(512+256)])
}


plot(sm.ref[1,],type=&#39;l&#39;,ylim=c(0.2,1),main=&#39;n=512&#39;,ylab=&#39;&#39;)
for(j in 2:6){
  if(j&gt;3){col=3}else{col=1}
  lines(sm.ref[j,],col=col)
}

plot(result[[1]]$x_t/result[[1]]$n_t,main=&#39;Normal1&#39;,col=&#39;grey80&#39;)
lines(sm.ref[1,])

legend(&#39;topleft&#39;,c(&#39;normal&#39;,&#39;cancer&#39;),col=c(1,3),lty=c(1,1))
paste(&#39;smallest x_t&#39;)
unlist(lapply(result, function(x){min(x$x_t)}))
paste(&#39;smallest n_t&#39;)
unlist(lapply(result, function(x){min(x$n_t)}))</code></pre>
<pre class="r"><code>result=list()
for (s in 1:6) {
  # Cancer 1 samples
BS1=BS.cancer.ex[,s]
BS1=BS1[-remove.idx]
# Obtain coverage
BS1.cov=getCoverage(BS1,type=&#39;Cov&#39;)
# Obtain possible methylation
BS1.M=getCoverage(BS1,type=&#39;M&#39;)
# choose 2kb windows. 
w=804
nw=floor(length(BS1)/w)
# Obtain binomial data n_t and x_t
n_t=c()
x_t=c()
for (i in 1:nw) {
  n_t[i]=sum(BS1.cov[(w*(i-1)+1):(w*i)])
  x_t[i]=sum(BS1.M[(w*(i-1)+1):(w*i)])
}

n_T=sum(BS1.cov[-w*nw])
x_T=sum(BS1.M[-w*nw])
n_t=c(n_t,n_T)
x_t=c(x_t,x_T)

#bio.sm=smash_gen_lite(x_t,ntri=n_t,dist_family = &#39;binomial&#39;)
result[[s]]=list(n_t=n_t,x_t=x_t)
}

sm.ref=c()
for (k in 1:6) {
  nn=reflect(result[[k]]$n_t,direct = &#39;both&#39;,len = c(512,512))
  xx=reflect(result[[k]]$x_t,direct = &#39;both&#39;,len = c(512,512))
  ss=smash_gen_lite(xx,ntri=nn,dist_family = &#39;binomial&#39;)
  sm.ref=rbind(sm.ref,ss[513:(1024+512)])
}


plot(sm.ref[1,],type=&#39;l&#39;,ylim=c(0.2,1),main=&#39;n=1024&#39;,ylab=&#39;&#39;)
for(j in 2:6){
  if(j&gt;3){col=3}else{col=1}
  lines(sm.ref[j,],col=col)
}

legend(&#39;topleft&#39;,c(&#39;normal&#39;,&#39;cancer&#39;),col=c(1,3),lty=c(1,1))
paste(&#39;smallest x_t&#39;)
unlist(lapply(result, function(x){min(x$x_t)}))
paste(&#39;smallest n_t&#39;)
unlist(lapply(result, function(x){min(x$n_t)}))</code></pre>
<p>What if remove all the no coverage area?</p>
<p>For normal 1:</p>
<pre class="r"><code>BS1=BS.cancer.ex[,s]
idx=which(getCoverage(BS1,type=&#39;Cov&#39;)==0)
BS1=BS1[-idx]
w=ceiling(length(BS1)/512)
nw=floor(length(BS1)/w)

n_t=c()
x_t=c()
for (i in 1:nw) {
  n_t[i]=sum(BS1.cov[(w*(i-1)+1):(w*i)])
  x_t[i]=sum(BS1.M[(w*(i-1)+1):(w*i)])
}

n_T=sum(BS1.cov[-w*nw])
x_T=sum(BS1.M[-w*nw])
n_t=c(n_t,n_T)
x_t=c(x_t,x_T)


nn=reflect(n_t,direct = &#39;both&#39;,len = c(256,256))
xx=reflect(x_t,direct = &#39;both&#39;,len = c(256,256))
ss=smash_gen_lite(xx,ntri=nn,dist_family = &#39;binomial&#39;)

plot(x_t/n_t,col=&#39;grey80&#39;)
lines(ss[257:(512+256)])</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] workflowr_1.1.1   Rcpp_0.12.18      digest_0.6.17    
 [4] rprojroot_1.3-2   R.methodsS3_1.7.1 backports_1.1.2  
 [7] git2r_0.23.0      magrittr_1.5      evaluate_0.11    
[10] stringi_1.2.4     whisker_0.3-2     R.oo_1.22.0      
[13] R.utils_2.7.0     rmarkdown_1.10    tools_3.5.1      
[16] stringr_1.3.1     yaml_2.2.0        compiler_3.5.1   
[19] htmltools_0.3.6   knitr_1.20       </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
