<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Dongyue Xie" />

<meta name="date" content="2018-09-29" />

<title>Literature review of smoothing non-Gaussian sequence data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">smashgen</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Literature review of smoothing non-Gaussian sequence data</h1>
<h4 class="author"><em>Dongyue Xie</em></h4>
<h4 class="date"><em>2018-09-29</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-11-14</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/DongyueXie/smash-gen/tree/777172185e414d6454db4529c80a2d48c95b02a8" target="_blank">7771721</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  analysis/chipexoeg.Rmd
    Untracked:  analysis/efsd.Rmd
    Untracked:  analysis/talk1011.Rmd
    Untracked:  data/chipexo_examples/
    Untracked:  data/chipseq_examples/
    Untracked:  talk.Rmd
    Untracked:  talk.pdf

Unstaged changes:
    Modified:   analysis/fda.Rmd
    Modified:   analysis/sigma.Rmd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/DongyueXie/smash-gen/blob/777172185e414d6454db4529c80a2d48c95b02a8/analysis/literature.Rmd" target="_blank">7771721</a>
</td>
<td style="text-align:left;">
Dongyue Xie
</td>
<td style="text-align:left;">
2018-11-14
</td>
<td style="text-align:left;">
wflow_publish(“analysis/literature.Rmd”)
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/DongyueXie/smash-gen/99b3deb594f72e0f5011a8c5e47c766025623432/docs/literature.html" target="_blank">99b3deb</a>
</td>
<td style="text-align:left;">
Dongyue Xie
</td>
<td style="text-align:left;">
2018-10-13
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/DongyueXie/smash-gen/blob/44e7e44ea8c8a752732a22b10b667d27f932350c/analysis/literature.Rmd" target="_blank">44e7e44</a>
</td>
<td style="text-align:left;">
Dongyue Xie
</td>
<td style="text-align:left;">
2018-10-13
</td>
<td style="text-align:left;">
add tf
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/DongyueXie/smash-gen/a21393a2ec5c7ef6125027bd8524dfc0f0ae70c1/docs/literature.html" target="_blank">a21393a</a>
</td>
<td style="text-align:left;">
Dongyue Xie
</td>
<td style="text-align:left;">
2018-10-05
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<p>This is a broad list of literature on smoothing Gaussian/non-Gaussian data.</p>
<div id="nonparametric-methods" class="section level2">
<h2>Nonparametric methods</h2>
<ol style="list-style-type: decimal">
<li><p>KNN</p></li>
<li><p>Kernel smootihng methods: Chapter 6 of <a href="https://web.stanford.edu/~hastie/ElemStatLearn/">ESL</a>.</p></li>
<li><p>Local regression(linear &amp; higher order): <a href="http://web.ipac.caltech.edu/staff/fmasci/home/astro_refs/LocalRegressionBook_1999.pdf">Loader, C. (2006). Local regression and likelihood. Springer Science &amp; Business Media.</a></p></li>
</ol>
<p>Note: Local regression makes no global assumptions about the function but assume that locally it can be well approximated with a member of a simple class of parametric function. Only observations in certain window are used.</p>
<ol start="4" style="list-style-type: decimal">
<li><p>Splines: regression splines, smoothing splines; More generally, reproducing kernel Hilbert space: Chapter 5 of <a href="https://web.stanford.edu/~hastie/ElemStatLearn/">ESL</a></p></li>
<li><p>Locally adaptive estimators: wavelet( <a href="https://www.di.ens.fr/~mallat/papiers/WaveletTourChap1-2-3.pdf">Mallat, S. (1999). A wavelet tour of signal processing. Elsevier</a> ), Locally adaptive regression splines(A varaiant of smooting splines achieves better local adaptivity. <a href="https://projecteuclid.org/euclid.aos/1034276635">Mammen, E., &amp; van de Geer, S. (1997). Locally adaptive regression splines. The Annals of Statistics, 25(1), 387-413.</a>), Trend filtering( <a href="http://web.stanford.edu/~boyd/papers/l1_trend_filter.html">Kim, S. J., Koh, K., Boyd, S., &amp; Gorinevsky, D. (2009). l_1 Trend Filtering. SIAM review, 51(2), 339-360.</a> ).</p></li>
<li><p>Additive models: Sparse additive models, Generalized additive mixed models.</p></li>
</ol>
<p>More on trend filtering and additive models:</p>
<p>Trend filtering:</p>
<p><a href="http://www.jmlr.org/papers/volume17/15-147/15-147.pdf">Wang, Y. X., Sharpnack, J., Smola, A. J., &amp; Tibshirani, R. J. (2016). Trend filtering on graphs. The Journal of Machine Learning Research, 17(1), 3651-3691.</a></p>
<p><a href="https://www.tandfonline.com/doi/abs/10.1080/10618600.2015.1054033">Ramdas, A., &amp; Tibshirani, R. J. (2016). Fast and flexible ADMM algorithms for trend filtering. Journal of Computational and Graphical Statistics, 25(3), 839-858.</a> The R package for this algo is glmgen.</p>
<p><a href="http://www.stat.cmu.edu/~ryantibs/papers/trendfilter.pdf">Tibshirani, R. J. (2014). Adaptive piecewise polynomial estimation via trend filtering. The Annals of Statistics, 42(1), 285-323.</a></p>
<p>Additive models:</p>
<p><a href="http://www.stat.cmu.edu/~ryantibs/papers/additivetf.pdf">Sadhanala, V., &amp; Tibshirani, R. J. (2017). Additive Models with Trend Filtering. arXiv preprint arXiv:1702.05037.</a></p>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5321231/">Petersen, A., Witten, D., &amp; Simon, N. (2016). Fused lasso additive model. Journal of Computational and Graphical Statistics, 25(4), 1005-1025.</a></p>
<p><a href="https://amstat.tandfonline.com/doi/full/10.1080/10618600.2015.1089775#.W8J4Ci2ZPfZ">Yin Lou, Jacob Bien, Rich Caruana &amp; Johannes Gehrke (2016) Sparse Partially Linear Additive Models, Journal of Computational and Graphical Statistics, 25:4, 1126-1140</a></p>
<p><a href="http://faculty.chicagobooth.edu/workshops/econometrics/pdf/pdf/Fall%202018/Shojaie,%20GSAM.pdf">Generalized Sparse Additive Models</a></p>
<p><a href="https://web.stanford.edu/~hastie/Papers/gamsel.pdf">Chouldechova, A., &amp; Hastie, T. (2015). Generalized additive model selection. arXiv preprint arXiv:1506.03850.</a></p>
</div>
<div id="exponential-family" class="section level2">
<h2>Exponential family</h2>
<p>Nonparametric regression for exponential family:</p>
<p><a href="https://arxiv.org/pdf/1010.3836.pdf">Brown, Lawrence D., T. Tony Cai, and Harrison H. Zhou. “Nonparametric regression in exponential families.” The annals of statistics 38.4 (2010): 2005-2046.</a></p>
<p><a href="https://www.jstor.org/stable/pdf/2290771.pdf?refreqid=excelsior%3A9899f3e009de18206db09c6eec5a069c">Cleveland, W. S., Mallows, C. L., &amp; McRae, J. E. (1993). ATS methods: Nonparametric regression for non-Gaussian data. Journal of the American Statistical Association, 88(423), 821-835.</a></p>
<p><a href="http://www3.stat.sinica.edu.tw/statistica/oldpdf/A16n317.pdf">Zhang, H. H., &amp; Lin, Y. (2006). Component selection and smoothing for nonparametric regression in exponential families. Statistica Sinica, 1021-1041.</a></p>
<p><a href="https://www.sciencedirect.com/science/article/pii/S0167715211002719">Bianco, A. M., Boente, G., &amp; Sombielle, S. (2011). Robust estimation for nonparametric generalized regression. Statistics &amp; Probability Letters, 81(12), 1986-1994.</a></p>
<p><a href="https://arxiv.org/pdf/1701.07263.pdf">Fryzlewicz, P. (2017). Likelihood ratio Haar variance stabilization and normalization for Poisson and other non-Gaussian noise removal. arXiv preprint arXiv:1701.07263.</a></p>
<p><a href="https://www.jstor.org/stable/pdf/2289465.pdf">Local Likelihood Estimation</a></p>
<p>Generalized addtive model</p>
<p><a href="http://www.stat.wisc.edu/~yandell/doc/1986/7.JASA.pdf">O’sullivan, F., Yandell, B. S., &amp; Raynor Jr, W. J. (1986). Automatic smoothing of regression functions in generalized linear models. Journal of the American Statistical Association, 81(393), 96-103.</a></p>
</div>
<div id="poisson" class="section level2">
<h2>Poisson</h2>
<p><a href="https://www.jstor.org/stable/2670007?seq=1#metadata_info_tab_contents">Kolaczyk, E. (1999). Bayesian Multiscale Models for Poisson Processes. Journal of the American Statistical Association, 94(447), 920-933. doi:10.2307/2670007</a></p>
<p><a href="http://stats.lse.ac.uk/fryzlewicz/Poisson/jcgs.pdf">Fryzlewicz, P., &amp; Nason, G. P. (2004). A Haar-Fisz algorithm for Poisson intensity estimation. Journal of computational and graphical statistics, 13(3), 621-638.</a></p>
<p><a href="https://ieeexplore.ieee.org/document/761328">Timmerman, K., &amp; Nowak, R. D. (1999). Multiscale modeling and estimation of Poisson processes with application to photon-limited imaging. IEEE Transactions on Information Theory, 45(3), 846-842.</a></p>
</div>
<div id="binomial" class="section level2">
<h2>Binomial</h2>
<p><a href="https://aip.scitation.org/doi/10.1063/1.1137498">Marchand, P., &amp; Marmet, L. (1983). Binomial smoothing filter: A way to avoid some pitfalls of least‐squares polynomial smoothing. Review of scientific instruments, 54(8), 1034-1041.</a></p>
<p><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2012-13-10-r83#Abs1">Hansen, K. D., Langmead, B., &amp; Irizarry, R. A. (2012). BSmooth: from whole genome bisulfite sequencing reads to differentially methylated regions. Genome biology, 13(10), R83.</a></p>
<p>Note: It actually uses local likelihood moother and assumes <span class="math inline">\(logit(\pi)\)</span> is approximated by a second degree polynomial. They assume that data follow a binomial distribution and the parameters defining the polynomial are estimated by fitting a weighted generalized linear model to the data inside the genomic window.</p>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
